.gitlab-ci-check-helm-version-bump.yml

```diff
@@ -0,0 +1,110 @@
+    # e.g.: for the mender-integration/extra/release_tool.py generate-delta-worker becames: generate_delta_worker:
+    # release_tool -m git ${CI_PROJECT_NAME} container
+    - |
+      echo "INFO - mapping source project to values.yaml project:"
+      case $CONTAINER in
+        generate-delta-worker)
+          VALUES_REF="generate_delta_worker"
+          ;;
+        *)
+          VALUES_REF_TMP=${CONTAINER#"alvaldi-"}  #removes prefix: alvaldi-
+          VALUES_REF=${VALUES_REF_TMP//-/_} #replaces - with _
+      esac
+      echo "DEBUG - container name inside values file: ${VALUES_REF}"
+    - |
+      echo "INFO - checking values files"
+      test -e ${CHART_DIR}/values-${SYNC_ENVIRONMENT}.yaml || ( echo "ERROR - ${CHART_DIR}/values-${SYNC_ENVIRONMENT}.yaml doesn't exists" ; exit 1 )
+      test -e ${CHART_DIR}/Chart.yaml || ( echo "ERROR - ${CHART_DIR}/Chart.yaml doesn't exists" ; exit 1 )
+    - |
+      echo "INFO - bumping version ${DOCKER_PUBLISH_COMMIT_TAG} to ${VALUES_REF} container"
+      THIS_KEY=".${VALUES_REF}.image.tag" THIS_VALUE="${DOCKER_PUBLISH_COMMIT_TAG}" yq -i 'eval(strenv(THIS_KEY)) = strenv(THIS_VALUE)' ${CHART_DIR}/values-${SYNC_ENVIRONMENT}.yaml
+      git add ${CHART_DIR}/values-${SYNC_ENVIRONMENT}.yaml
+    - |
+      echo "DEBUG - container registry is: $CONTAINER_REGISTRY"
+      if [[ -n "${CONTAINER_REGISTRY}" ]]; then
+        echo "INFO - bumping registry ${CONTAINER_REGISTRY} to ${VALUES_REF} container"
+        THIS_KEY=".${VALUES_REF}.image.registry" THIS_VALUE="${CONTAINER_REGISTRY}" yq -i 'eval(strenv(THIS_KEY)) = strenv(THIS_VALUE)' ${CHART_DIR}/values-${SYNC_ENVIRONMENT}.yaml
+        git add ${CHART_DIR}/values-${SYNC_ENVIRONMENT}.yaml
+      fi
+    - |
+      echo "DEBUG - container repository is: $CONTAINER_REPOSITORY"
+      if [[ -n "${CONTAINER_REPOSITORY}" ]]; then
+        echo "INFO - bumping repository ${CONTAINER_REPOSITORY} to ${VALUES_REF} container"
+        THIS_KEY=".${VALUES_REF}.image.repository" THIS_VALUE="${CONTAINER_REPOSITORY}" yq -i 'eval(strenv(THIS_KEY)) = strenv(THIS_VALUE)' ${CHART_DIR}/values-${SYNC_ENVIRONMENT}.yaml
+        git add ${CHART_DIR}/values-${SYNC_ENVIRONMENT}.yaml
+      fi
+    - |
+      echo "INFO - bumping helm chart version"
+      FULL_VERSION=$(yq ".version" ${CHART_DIR}/Chart.yaml)
+      MAJOR_VERSION=$(echo $FULL_VERSION | cut -f1 -d.)
+      MINOR_VERSION=$(echo $FULL_VERSION | cut -f2 -d.)
+      THIS_VALUE="${MAJOR_VERSION}.${MINOR_VERSION}.${HELM_PATCH_VERSION}" yq -i '.version = strenv(THIS_VALUE)' ${CHART_DIR}/Chart.yaml
+      git add ${CHART_DIR}/Chart.yaml
+    # Commit
+    - git commit -sm "[CI/CD] bump helm chart"
+    # Push back (5 retries)
+    - |
+      for retry in $(seq 5); do
+        if git push github overlay-version-bump:${MAIN_BRANCH:-master}; then
+          exit 0
+        fi
+        git fetch github ${MAIN_BRANCH:-master}
+        git rebase github/${MAIN_BRANCH:-master}
```

Comment:
If this pipeline is run in parallel, this will likely cause a merge conflict.

Reply:
That's true. In that case I would prefer solving the conflicts by hand, to be able to decide which version wins for production. For now I have no better idea. Let's improve it along the way